specs/plugin-extensibility.md

```diff
@@ -22,13 +22,37 @@ Keys and associated certificates used for signing artifacts using Notation could
 
 Notation will invoke plugins as executable, pass parameters using command line arguments, and use standard IO streams to pass request/response payloads. This mechanism is used as Go language (used to develop [Notation Go library](https://github.com/notaryproject/notation-go)) does not have a [in built support](https://github.com/golang/go/issues/19282) to load and execute plugins that works across OS platforms. Other mechanisms like gRPC require every plugin to be implemented as a service/daemon.
 
+### Package and release a plugin
+
+Notation supports the installation of a plugin from an `https` URL or from a file in the filesystem. To support the plugin installation and management using `notation plugin` commands, the plugin publisher must adhere to the following conventions:
+
+* Plugin executable file MUST follow the naming convention `notation-{plugin-name}`. On `windows` OS, the file extension `.exe` is REQUIRED.
+* The plugin distribution format MUST be either .zip, .tar.gz, or a single plugin executable file. If the release format is _single plugin executable file_, it is highly recommended to compress it into an archive for installation efficiency consideration.
+* If the archive format is `.zip` or `.tar.gz`, there MUST be one and only one plugin executable file within each archive.
+* Plugin publisher SHOULD provide SHA256 checksum for each released archive and executable file if they want to enable users to install a plugin from an `https` URL.
```

Comment:
1. Might be an ambiguity: there is plugin executable file inside an archive as well. In this case, only SHA256sum of the archive is required, the publisher does not need to provide SHA256sum of the executable file. 
2. Since this guide is under ```To support the plugin installation and management using `notation plugin` commands, the plugin publisher must adhere to the following conventions:```, we should update the `SHOULD` to `MUST`, otherwise, the installation would fail.
```suggestion
* Plugin publisher MUST provide SHA256 checksum for each released archive or _single plugin executable file_ if they want to enable users to install a plugin from an `https` URL.
```